src/main/java/io/javalin/core/util/MethodNotAllowedUtil.kt

```diff
@@ -0,0 +1,47 @@
+package io.javalin.core.util
+
+import io.javalin.core.HandlerType
+import io.javalin.core.PathMatcher
+import io.javalin.translator.json.JavalinJsonPlugin.objectToJsonMapper
+
+private data class JsonMethodNotAllowed(val availableMethods: List<HandlerType>)
+
+object MethodNotAllowedUtil {
+
+    @JvmStatic
+    fun findAvailableHandlerTypes(matcher: PathMatcher, requestUri: String): List<HandlerType> {
```

Comment:
```
enumValues<HandlerType>().flatMap { matcher.findEntries(it, requestUri)  }
``` 
Maybe I'm biased, but this looks more readable to me

Reply:
You can also use an extension function on `PathMatcher` instead of creating a util
```kotlin
fun PathMatcher.findAvailableHandlerTypes(requestUri: String): List<HandlerType>
```

Then it can be used in `JavalinServlet` like this:
```kotlin
val availableHandlerTypes = matcher.findAvailableHandlerTypes(requestUri)
```